### ISSUE TITLE
Running multiple times in parallel against a server causes weird errors

### ISSUE THREAD

**Issue by dpp23** [nvuillam/npm-groovy-lint#215]
When running two (or more) instances of `npm-groovy-lint` in parallel against a localhost codenarc server, I've hit various weird errors:

1. Empty response from the server (?)
```
Unable to use CodeNarc JSON result
 {"err":"Unable to parse INIT: Hit from Host: localhost on port: 59078\nREQUEST BODY: {<redacted>"}
Unexpected error: Cannot convert undefined or null to object
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at NpmGroovyLintFix.run (node_modules/npm-groovy-lint/lib/groovy-lint-fix.js:51:31)
    at NpmGroovyLint.postProcess (node_modules/npm-groovy-lint/lib/groovy-lint.js:280:34)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async NpmGroovyLint.run (ode_modules/npm-groovy-lint/lib/groovy-lint.js:66:13)
    at async node_modules/npm-groovy-lint/lib/index.js:12:9
```

2. Getting an incorrect result and trying to parse the wrong files (?)
```
Unexpected error: ENOENT: no such file or directory, open 'foo'
Error: ENOENT: no such file or directory, open 'foo'
```

Where `foo` is actually the name of a file that is from the other invocation.

Passing `--noserver` fixes these, but then everything ends up quite slow.

I appreciate that this project needs maintainers, but wanted to write this issue down so people know why these weird errors occur.

An aside: Is there a way to just hand `npm-groovy-lint` a precomputed Codenarc report and avoid hitting Codenarc at all? In my use case, I am already in a Java (Gradle) context, so running Codenarc is very fast and efficient. I want to use `npm-groovy-lint` for its ability to fix issues, so I end up executing `npm-groovy-lint` from Java just so it starts another JVM in order to call Codenarc.

**Comment by nvuillam** [comment]
Mmmm why using several npm-groovy-lint in parallel ?
I don't remember exactly how I managed parallelism... I have to check :/ (but not a lot of free time these days :/ )

**Comment by dpp23** [comment]
> Mmmm why using several npm-groovy-lint in parallel ?

My use case might be an uncommon one - I am writing a gradle plugin that utilizes `npm-groovy-lint` in order to fix issues found by codenarc in the gradle groovy code. As the plugin is added to each gradle subproject, the lint end up running once per project and hence several invocations in parallel.
